### PR TITLE
Using token metadata client sdk

### PR DESCRIPTION
**Pull request by febo**
This PR switches the Token Metadata crate to use the Rust client SDK instead of the program crate.

### CODE REVIEW

File: spl/src/metadata.rs [hunk 1]
```diff
@@ -786,29 +805,25 @@ impl Deref for MetadataAccount {
...
-impl MasterEditionAccount {
-    pub const LEN: usize = mpl_token_metadata::state::MAX_MASTER_EDITION_LEN;
-}
+pub struct MasterEditionAccount(mpl_token_metadata::accounts::MasterEdition);
 
 impl anchor_lang::AccountDeserialize for MasterEditionAccount {
     fn try_deserialize(buf: &mut &[u8]) -> anchor_lang::Result<Self> {
         let me = Self::try_deserialize_unchecked(buf)?;
-        if me.key != mpl_token_metadata::state::MasterEditionV2::key() {
+        if me.key != mpl_token_metadata::types::Key::MasterEditionV2 {
```
Comment (acheroncrypto) [inline spl/src/metadata.rs:813]: Are the changes for the `MasterEditionAccount` correct? The account changed from `MasterEditionV2` to `MasterEdition` but we are checking the key of `MasterEditionV2` here.

https://github.com/coral-xyz/anchor/blob/a465ae6fb49cc9058f4674ae40a82b91306ef0b2/spl/src/metadata.rs#L808

**Review by acheroncrypto — COMMENTED** [review]
Thanks. This looks good to me overall but we don't have tests for these so might be missing something given the changes. Any chance you've tested these changes?

This also looks like a breaking change, would be great if you could you add an entry to the [CHANGELOG](https://github.com/coral-xyz/anchor/blob/master/CHANGELOG.md#breaking).

File: spl/src/metadata.rs (hunk 1 above)
Comment (febo) [reply spl/src/metadata.rs:813]: Yes, the account struct `MasterEdition` is compatible with both `V1` and `v2`. And the test is equivalent – i.e., `MasterEditionV2::key()` gives the same value as `Key::MasterEditionV2`.

**Comment by febo** [comment]
Added an entry on the `CHANGELOG`. I have not tested the changes on the anchor types, as I was relying on the existing ones.

**Review by acheroncrypto — APPROVED** [review]
Thanks a lot!

**Pull request merged**